cvmfs/magic_xattr.cc

```diff
@@ -492,33 +556,77 @@ static std::string ListProxy(download::DownloadManager *dm) {
       buf += proxy_chain[i][j].url;
       buf += "\n";
     }
+
+    if (buf.size() > BaseMagicXattr::kMaxCharsPerPage) {
+      result_pages->push_back(buf);
+      buf = "";
+    }
+  }
+
+  if (buf.size() > 0) {
+    result_pages->push_back(buf);
   }
-  return buf;
 }
 
-std::string ProxyListMagicXattr::GetValue() {
-  return ListProxy(xattr_mgr_->mount_point()->download_mgr());
+void ProxyListMagicXattr::FinalizeValue() {
+  ListProxy(xattr_mgr_->mount_point()->download_mgr(), &result_pages_);
 }
 
-std::string ProxyListExternalMagicXattr::GetValue() {
-  return ListProxy(xattr_mgr_->mount_point()->external_download_mgr());
+void ProxyListExternalMagicXattr::FinalizeValue() {
+  ListProxy(xattr_mgr_->mount_point()->external_download_mgr(), &result_pages_);
 }
 
 bool PubkeysMagicXattr::PrepareValueFenced() {
-  pubkeys_ = xattr_mgr_->mount_point()->signature_mgr()->GetActivePubkeys();
+  pubkeys_ = xattr_mgr_->mount_point()->signature_mgr()
+                                                   ->GetActivePubkeysAsVector();
   return true;
 }
 
-std::string PubkeysMagicXattr::GetValue() {
-  return pubkeys_;
+void PubkeysMagicXattr::FinalizeValue() {
+  size_t full_size = 0;
+
+  for (size_t i = 0; i < pubkeys_.size(); i++) {
+    full_size += pubkeys_[i].size();
+  }
+
+  if (full_size == 0) {
+    return;
+  }
+
+  if (full_size <= kMaxCharsPerPage) {
+    std::string res = "";
+
+    for (size_t i = 0; i < pubkeys_.size(); i++) {
+      res += pubkeys_[i] + "\n";
+    }
+
+    result_pages_.push_back(res);
+  } else {
```

Comment:
It seems like we don't need the `if` branch. This `else` branch already covers all cases.